code/game/objects/structures/janicart.dm

```diff
@@ -78,7 +78,7 @@
 			update_icon()
 		else
 			user << "<span class='warning'>[src] can't hold any more signs!</span>"
-	else if(mybag)
+	else if(mybag && !istype(I, /obj/item/weapon/reagent_containers))
```

Comment:
can't dump food, can't dump grown stuff, etc...
what you're looking for is !I.is_open_container()